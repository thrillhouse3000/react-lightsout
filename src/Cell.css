.Cell {
  height: 100px;
  width: 100px;
  background-color: rgb(50, 50, 50);
  cursor: pointer;
  border-radius: 10px;
}

.Cell-lit {
  background-color: white;
  box-shadow: rgba(252, 255, 169, 0.795) 0px 0px 29px 0px;
}