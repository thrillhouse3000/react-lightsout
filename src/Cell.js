import React from "react";
import "./Cell.css";

/** A single cell on the board.
 *
 * This has no state --- just two props:
 *
 * - flipCellsAroundMe: a function rec'd from the board which flips this
 *      cell and the cells around of it
 *
 * - isLit: boolean, is this cell lit?
 *
 * This handles clicks --- by calling flipCellsAroundMe
 *
 **/

function Cell({ flipCellsAroundMe, isLit, coord }) {
  const classes = `Cell ${isLit ? "Cell-lit" : ""}`;
  return <td data-testid={coord} className={classes} onClick={() => flipCellsAroundMe(coord)}></td>;
}

export default Cell;
