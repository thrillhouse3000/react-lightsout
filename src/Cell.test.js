import { render } from '@testing-library/react';
import Cell from './Cell'

it('renders without crashing', () => {
    render(<table><tbody><tr><Cell/></tr></tbody></table>);
})

it('should match snapshot', () => {
    const {asFragment} = render(<table><tbody><tr><Cell className='Cell'/></tr></tbody></table>);
    expect(asFragment()).toMatchSnapshot()  
})

it('should match snapshot', () => {
    const {asFragment} = render(<table><tbody><tr><Cell className='Cell Cell-lit'/></tr></tbody></table>);
    expect(asFragment()).toMatchSnapshot()  
})