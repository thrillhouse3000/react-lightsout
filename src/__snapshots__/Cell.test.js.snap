// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell "
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`should match snapshot 2`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell "
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
